From eb58757bd8c776da56c0a23559e7c7ba7a01b522 Mon Sep 17 00:00:00 2001
From: aoleary <seanm187@gmail.com>
Date: Wed, 22 Dec 2021 14:07:01 +0000
Subject: [PATCH] g4-common: camera fixed by https://github.com/DevUt

---
 libc/bionic/pthread_internal.cpp | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

diff --git a/libc/bionic/pthread_internal.cpp b/libc/bionic/pthread_internal.cpp
index 6fddefe43..7fdbf3dff 100644
--- a/libc/bionic/pthread_internal.cpp
+++ b/libc/bionic/pthread_internal.cpp
@@ -100,18 +100,10 @@ pthread_internal_t* __pthread_internal_find(pthread_t thread_id, const char* cal
     }
   }
 
-  // Historically we'd return null, but
-  if (android_get_application_target_sdk_version() >= __ANDROID_API_O__) {
     if (thread == nullptr) {
-      // This seems to be a common mistake, and it's relatively harmless because
-      // there will never be a valid thread at address 0, whereas other invalid
-      // addresses might sometimes contain threads or things that look enough like
-      // threads for us to do some real damage by continuing.
-      // TODO: try getting rid of this when Treble lets us keep vendor blobs on an old API level.
-      async_safe_format_log(ANDROID_LOG_WARN, "libc", "invalid pthread_t (0) passed to %s", caller);
+      async_safe_format_log(ANDROID_LOG_WARN, "libc", "hacked invalid pthread_t(0) passed to %s",caller);
     } else {
-      async_safe_fatal("invalid pthread_t %p passed to %s", thread, caller);
+      async_safe_format_log(ANDROID_LOG_WARN,"libc"," hacked invalid pthread_t %p passed to %s", thread, caller);
     }
-  }
   return nullptr;
 }
